src/jit/hwintrinsicxarch.cpp

```diff
@@ -204,8 +231,10 @@ GenTree* Compiler::impX86HWIntrinsic(NamedIntrinsic intrinsic, CORINFO_METHOD_HA
     // Will throw PlatformNotSupportedException if
     // - calling hardware intrinsics on unsupported hardware
     // - calling SIMD hardware intrinsics with featureSIMD=false
-    if ((!compSupports(isa) || (!featureSIMD && isa != InstructionSet_BMI1 && isa != InstructionSet_BMI2 &&
-                                isa != InstructionSet_LZCNT && isa != InstructionSet_POPCNT)) &&
+    // - the ISA class is not yet fully implemented and EnableIncompleteISAClass=false
+    if ((!compSupports(isa) || (!JitConfig.EnableIncompleteISAClass() && !isFullyImplmentedISAClass(isa)) ||
```

Comment:
I think it would be cleaner to have this check look something like this:
```
if ((!compSupports(isa) || !compSupportsIntrinsics(isa)) && !isIntrinsicAnIsSupportedPropertyGetter(intrinsic))
```
Then, `compSupportsIntrinsics(isa)` can incorporate both the check for those that are fully supported, as well as, under `DEBUG` only, checking `JitConfig.EnableIncompleteISAClass()`.

Reply:
That would be more elegant. Will change.